src/components/sidebar/index.tsx

```diff
@@ -45,14 +47,31 @@ const addActiveStateMetadata = (
 
     if (foundActiveItem) {
       itemCopy[item.routes ? 'hasActiveChild' : 'isActive'] = false
-    } else if (item.routes) {
-      const result = addActiveStateMetadata(currentPath, item.routes)
+    }
+
+    if (item.routes) {
+      const result = addItemMetadata(currentPath, item.routes)
       foundActiveItem = result.foundActiveItem
       itemCopy.routes = result.itemsWithMetadata
       itemCopy.hasActiveChild = result.foundActiveItem
-    } else {
+      itemCopy.id = `submenu-${itemCopy.title
```

Comment:
Nit: I've seen [`slugify` ](https://www.npmjs.com/package/slugify) in use in a few of our projects, might be useful here.

Reply:
I actually tried that one! But it doesn't remove `/` so I decided not to use it.